src/main/java/org/openstreetmap/atlas/checks/validation/points/DuplicateNodeCheck.java

```diff
@@ -39,7 +40,7 @@ public DuplicateNodeCheck(final Configuration configuration)
     @Override
     public boolean validCheckForObject(final AtlasObject object)
     {
-        return object instanceof Node && !this.isFlagged(((Node) object).getLocation());
```

Comment:
This does accomplish flagging all overlapping nodes, but it would be better if there was only one flag per location. That way if someone was fixing these issues they were not seeing the same problem multiple times. 
All the Nodes in a location could be in the same flag, or just the same single Node for each location (highest or lowest identifier perhaps).